### PR TITLE
[pylint-01] small pylint fixes

### PR DESCRIPTION
**Pull request by nth10sd**
Testing via pylint is not turned on yet, but I'd thought to break up the pylint stuff into smaller PRs, so here is one of them.

### CODE REVIEW

File: util/tooltool/tooltool.py [hunk 1]
```diff
@@ -361,7 +358,7 @@ def list_manifest(manifest_file):
     try:
         manifest = open_manifest(manifest_file)
     except InvalidManifest as e:
-        log.error("failed to load manifest file at '%s': %s" % (
+        log.error("failed to load manifest file at '%s': %s", (
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: The brackets should not be around the args following comma (`"" % (a1, a2)` vs. `log.error("", a1, a2)`)

File: util/tooltool/tooltool.py [hunk 2]
```diff
@@ -379,7 +376,7 @@ def validate_manifest(manifest_file):
...
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 3]
```diff
@@ -392,10 +389,9 @@ def validate_manifest(manifest_file):
         else:
             if not f.validate():
                 invalid_files.append(f)
-    if len(invalid_files + absent_files) == 0:
+    if not invalid_files + absent_files:
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: This is correct but not as clear as `if not (invalid_files + absent_files):` (I had to look up the precedence to confirm)

File: util/tooltool/tooltool.py [hunk 4]
```diff
@@ -554,7 +549,7 @@ def fetch_files(manifest_file, base_urls, filenames=[], cache_folder=None,
     try:
         manifest = open_manifest(manifest_file)
     except InvalidManifest as e:
-        log.error("failed to load manifest file at '%s': %s" % (
+        log.error("failed to load manifest file at '%s': %s", (
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 5]
```diff
@@ -588,15 +583,15 @@ def fetch_files(manifest_file, base_urls, filenames=[], cache_folder=None,
...
+                         "and try to fetch it", f.filename)
                 os.remove(os.path.join(os.getcwd(), f.filename))
 
         # check if file is already in cache
         if cache_folder and f.filename not in present_files:
             try:
                 shutil.copy(os.path.join(cache_folder, f.digest),
                             os.path.join(os.getcwd(), f.filename))
-                log.info("File %s retrieved from local cache %s" %
+                log.info("File %s retrieved from local cache %s",
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 6]
```diff
@@ -610,11 +605,11 @@ def fetch_files(manifest_file, base_urls, filenames=[], cache_folder=None,
                     # the file copied from the cache is invalid, better to
                     # clean up the cache version itself as well
                     log.warn("File %s retrieved from cache is invalid! I am deleting it from the "
-                             "cache as well" % f.filename)
+                             "cache as well", f.filename)
                     os.remove(os.path.join(os.getcwd(), f.filename))
                     os.remove(os.path.join(cache_folder, f.digest))
             except IOError:
-                log.info("File %s not present in local cache folder %s" %
+                log.info("File %s not present in local cache folder %s",
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 7]
```diff
@@ -652,7 +647,7 @@ def fetch_files(manifest_file, base_urls, filenames=[], cache_folder=None,
         if filerecord_for_validation.validate():
             # great!
             # I can rename the temp file
-            log.info("File integrity verified, renaming %s to %s" %
+            log.info("File integrity verified, renaming %s to %s",
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 8]
```diff
@@ -663,22 +658,22 @@ def fetch_files(manifest_file, base_urls, filenames=[], cache_folder=None,
...
                     if not os.path.exists(cache_folder):
-                        log.info("Creating cache in %s..." % cache_folder)
+                        log.info("Creating cache in %s...", cache_folder)
                         os.makedirs(cache_folder, 0700)
                     shutil.copy(os.path.join(os.getcwd(), localfile.filename),
                                 os.path.join(cache_folder, localfile.digest))
-                    log.info("Local cache %s updated with %s" % (cache_folder,
-                                                                 localfile.filename))
+                    log.info("Local cache %s updated with %s", (cache_folder,
+                                                                localfile.filename))
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 9]
```diff
@@ -663,22 +658,22 @@ def fetch_files(manifest_file, base_urls, filenames=[], cache_folder=None,
...
                     touch(os.path.join(cache_folder, localfile.digest))
                 except (OSError, IOError):
-                    log.warning('Impossible to add file %s to cache folder %s' %
+                    log.warning("Impossible to add file %s to cache folder %s",
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 10]
```diff
@@ -732,27 +726,27 @@ def purge(folder, gigs):
...
             break
 
 
 def _log_api_error(e):
     if hasattr(e, 'hdrs') and e.hdrs['content-type'] == 'application/json':
         json_resp = json.load(e.fp)
-        log.error("%s: %s" % (json_resp['error']['name'],
-                              json_resp['error']['description']))
+        log.error("%s: %s", (json_resp['error']['name'],
+                             json_resp['error']['description']))
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

File: util/tooltool/tooltool.py [hunk 11]
```diff
@@ -897,9 +891,9 @@ def process_command(options, args):
     start doing the right thing with them"""
     cmd = args[0]
     cmd_args = args[1:]
-    log.debug("processing '%s' command with args '%s'" %
+    log.debug("processing '%s' command with args '%s'",
```
Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Ditto re: logging args in brackets.

**Comment by nth10sd** [comment]
The errors I partially fixed here (e.g. logging), were based on me running pylint locally and arbitrarily fixing the easy/easier ones.

**Comment by nth10sd** [comment]
Also, yes I'm looking at removing downloadBuild (and hence tooltool.py) but only once fuzzfetch gains feature parity with downloadBuild. (ditto autoBisect with the version from @pyoor) In the meantime, I'm fixing up whatever files we have in funfuzz for now.

File: util/tooltool/tooltool.py (hunk 3 above)
Comment (nth10sd) [inline util/tooltool/tooltool.py]: The specific pylint complaint is `Unnecessary parens after u'not' keyword (superfluous-parens)`

Comment (jschwartzentruber) [inline util/tooltool/tooltool.py]: Oh well 🤷‍

**Pull request merged**